Solutions/MailRisk/Package/mainTemplate.json

```diff
@@ -0,0 +1,316 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "Location for all resources. Default is the location of the resource group."
+            }
+        },
+        "FunctionName": {
+            "defaultValue": "MailRisk",
+            "minLength": 3,
+            "maxLength": 11,
+            "type": "string",
+            "metadata": {
+                "description": "Name of the function app that will be created. Must be unique globally."
+            }
+        },
+        "WorkspaceId": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Workspace ID and Primary Key can be found in the Azure Portal, and you should find the workspace that is connected to Microsoft Sentinel. Typically found under Log Analytics workspace -> Agents management"
+            }
+        },
+        "WorkspaceKey": {
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Workspace ID and Primary Key can be found in the Azure Portal, and you should find the workspace that is connected to Microsoft Sentinel. Typically found under Log Analytics workspace -> Agents management"
+            }
+        },
+        "APIKey": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "The Secure Practice API key and secret are created in the <a href=\"https://manage.securepractice.co/settings/security\" target=\"_blank\">settings in the admin portal</a>."
+            }
+        },
+        "APISecret": {
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "The Secure Practice API key and secret are created in the <a href=\"https://manage.securepractice.co/settings/security\" target=\"_blank\">settings in the admin portal</a>."
+            }
+        }
+    },
+    "variables": {
+        "FunctionName": "[concat(toLower(parameters('FunctionName')), uniqueString(resourceGroup().id))]",
```

Comment:
Solution id not available 
please create the package using v2 tool

Reply:
I manually added the mainTemplate.json last time as we do not have an arm template specific for the data connector, should I add that first, before repackaging with the tool?